includes/admin/class-wc-admin-addons.php

```diff
@@ -40,36 +40,97 @@ public static function get_featured() {
 	}
 
 	/**
-	 * Get sections for the addons screen
+	 * Build transient name
 	 *
-	 * @return array of objects
+	 * @param  string $category
+	 * @param  string $term
+	 * @param  string $country
+	 *
+	 * @return string transient name
 	 */
-	public static function get_sections() {
-		if ( false === ( $sections = get_transient( 'wc_addons_sections' ) ) ) {
-			$raw_sections = wp_safe_remote_get( 'https://d3t0oesq8995hv.cloudfront.net/addon-sections.json', array( 'user-agent' => 'WooCommerce Addons Page' ) );
-			if ( ! is_wp_error( $raw_sections ) ) {
-				$sections = json_decode( wp_remote_retrieve_body( $raw_sections ) );
+	public static function build_transient_name( $category, $term, $country ) {
+		$transient_name = 'wc_addons_extensions';
+		if ( $category ) {
+			$transient_name = $transient_name . '_category_' . $category;
+		}
+		if ( $term ) {
+			$transient_name = $transient_name . '_term_' . $term;
+		}
+		if ( $country ) {
+			$transient_name = $transient_name . '_country_' . $country;
+		}
+		return $transient_name;
+	}
+
+	/**
+	 * Build url parameter string
+	 *
+	 * @param  string $category
+	 * @param  string $term
+	 * @param  string $country
+	 *
+	 * @return string url parameter string
+	 */
+	public static function build_parameter_string( $category, $term, $country ) {
+
+		$paramters = array(
+			'category' => $category,
+			'term' => $term,
+			'country' => $country,
+		);
+
+		return '?' . http_build_query( $paramters );
+	}
 
-				if ( $sections ) {
-					set_transient( 'wc_addons_sections', $sections, WEEK_IN_SECONDS );
+	/**
+	 * Call API to get extensions
+	 *
+	 * @param  string $category
+	 * @param  string $term
+	 * @param  string $country
+	 *
+	 * @return array of extensions
+	 */
+	public static function get_extension_data( $category, $term, $country ) {
+		$transient_name = self::build_transient_name( $category, $term, $country );
+		$parameters = self::build_parameter_string( $category, $term, $country );
+		$addons = get_transient( $transient_name );
+		if ( false === ( $addons ) ) {
+			$raw_extensions = wp_remote_get(
+				'https://woocommerce.com/wp-json/wccom-extensions/1.0/search' . $parameters
+			);
+			if ( ! is_wp_error( $raw_extensions ) ) {
+				$addons = json_decode( wp_remote_retrieve_body( $raw_extensions ) )->products;
+				if ( $addons ) {
+					set_transient( $transient_name, $addons, WEEK_IN_SECONDS );
```

Comment:
We should be caching failures here too, for when WooCommerce.com is down.

Also, I'm not completely sold on caching results for actual search terms on the client side. I see the benefit of caching some things on the WooCommerce.com API side, when multiple different users with different sites search for the same thing in the same country and category.

But for *this* specific client-side cache to be effective, the user will have to go to the Extensions screen and perform a search. Then within a week, go to the Extensions screen and perform the exact same search. Sounds a bit too coincidental to me.

Caching categories is ok I feel, although if you think about it, what if a hot new featured extension is launched on WooCommerce.com, it has all the greatest features and all these other amazing things, and as a promo it costs only 7 cents, and we're running the promo for a week only. Some unlucky users will miss out on this great opportunity because of stale cache :)